### PR TITLE
Fix: Add custom annotation add for application edit page

### LINKED ISSUE #665: [Feature] Custom annotation add for application edit page
**Is your feature request related to a problem? Please describe.**
The application edit page offers editing application labels, in addition to the label is it possible to add a custom annotation area for the application?

**Describe the solution you'd like**
Custom annotation input area to add key-value pair.
Screen Shot 2023-01-17 at 22 07 00

### PR DESCRIPTION
**Pull request by yikuaibro**
### Description of your changes

Fixes #665 

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary.
- [ ] Run `yarn lint` to ensure the frontend changes are ready for review.
- [ ] Run `make reviewable`to ensure the server changes are ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: packages/velaux-ui/src/locals/Zh/zh.json [hunk 1]
```diff
@@ -102,6 +102,7 @@
   "Addon in extension": "",
   "Addon in management": "管理插件",
   "Alias": "别名",
+  "Annotations": "注释",
```
Comment (wonderflow) [inline packages/velaux-ui/src/locals/Zh/zh.json:105]:
```suggestion
  "Annotations": "注解",
```

Comment (wonderflow) [inline packages/velaux-ui/src/locals/Zh/zh.json:105]:
```suggestion
  "Annotations": "注解",
```

Comment (yikuaibro) [reply packages/velaux-ui/src/locals/Zh/zh.json]: done

**Comment by barnettZQG** [comment]
@yikuaibro Thanks for your contribution, please test it in your local environment and write the e2e test case.

File: pkg/server/domain/model/application.go [hunk 2]
```diff
@@ -39,6 +39,7 @@ type Application struct {
 	Description string            `json:"description"`
 	Icon        string            `json:"icon"`
 	Labels      map[string]string `json:"labels,omitempty"`
+	Annotations map[string]string `json:"annotations,omitempty"`
```
Comment (barnettZQG) [inline pkg/server/domain/model/application.go:42]: Let’s assign the annotations to application CR when deploying it. And make sure don’t override the inner annotations.

Comment (yikuaibro) [reply pkg/server/domain/model/application.go:42]: So need to consider annotations when creating applications? Does 'inner annotations' refer to the 'annotations' attribute of CR metadata?

**Comment by yikuaibro** [comment]
Hi @barnettZQG  , I hope everything is going well. I just wanted to send a friendly reminder about the pull request I submitted earlier and see if you had a chance to review it yet. If there's anything you need from me or if you have any feedback, please let me know. I appreciate your help and time.

File: pkg/server/domain/model/application.go (hunk 2 above)
Comment (barnettZQG) [reply pkg/server/domain/model/application.go:42]: Now, you only added the annotation field for the application DB model. After you deploy an application, you can use the `vela` CLI command to view the application CR. It is a real request feature that we could find the annotations in the application CR. Do you think so?

What is the inner annotation? Like the `app.oam.dev/publishVersion` is an inner annotation. It is generated by the system.

Comment (barnettZQG) [reply pkg/server/domain/model/application.go:42]: @yikuaibro

**Pull request closed without merging**